### ISSUE TITLE
Implement operations for RVResults

### ISSUE THREAD

**Issue by dlsun** [dlsun/symbulate#73]
Right now, the following code does not work:

```
X = RV(Exponential(rate=1/4))
x = X.sim(10000)
mean((x - x.mean()) ** 2)
```

That's because `x - x.mean()` is not well-defined. We would need to define operations on `RVResults` objects to make this work.

**Comment by kevindavisross** [comment]
I've attached a notebook with some tests of operations in RV and simulation space. 
[Simulation space operations.pdf](https://github.com/dlsun/symbulate/files/2700348/Simulation.space.operations.pdf)
 RV space appears to be working fine - aside from tuples created with *.  For example, in order to implement
```
X = RV(BivariateNormal() * Poisson(1))
(X + ((0, -10.4), 100)).draw()
```
I need to use
```
X, Y = RV(BivariateNormal() * Poisson(1))
((X + (0, -10.4)) & (Y + 100)).draw()
```

But I think **the bigger issues are with simulation space**.  In short,  things like `(X + 3).sim(10)` work fine, but things like `X.sim(10) + 3` return errors (though things like `X.draw() + 10` do work).  (`apply` seems to work fine in both RV and sim space.)

Before making any changes, I think we should think about this issue more carefully.  I think the idea from the beginning has been that whatever can be done in RV space should also have a sim space version, e.g. `X.apply(sqrt).sim(10)` and `X.sim(10).apply(sqrt)` are both possible.  My question is **do we really want the same operations to work in RV and sim spaces?**  I haven't thought this through, but the answer might be no.  

From a pedagogical perspective, I tell my students to first define all elements of the model - prob space, RVs, transformations - and then simulate.  Basically, distinguish between what one repetition of the simulation entails and summarizing the results of many simulations.  So I would encourage students to code as `X.apply(sqrt).sim(10)` rather than `X.sim(10).apply(sqrt)`.  Is there a pedagogical advantage to allowing the simulation space operation (other than just preventing coding errors)?  For example, does simulation space syntax `X.sim(10).apply(sqrt)` help students better understand LOTUS?

For single RVs, simulation space operations like `X.sim(10) + 3` seem natural, but it gets trickier when dealing with multiple RVs, like 
```
X = RV(BivariateNormal())
X.sim(10) + Vector((0, -10.4))
```
Another example: do we really want/need a simulation space equivalent of the following?
```
X, Y = RV(BivariateNormal(mean1=0,mean2=10) * BivariateNormal(mean1=100, mean2=20))
(X + Y).sim(10)
```

If we think of outcomes as rows and RVs as columns, then RV space operations apply row by row, outcome by outcome.  Should we force users to define in RV space anything that can be defined in RV space?  And then treat simulation space operations like vectorized numpy arrays operating on columns?  The latter might make things like `mean((x - x.mean()) ** 2)` and lots more easier/more natural.

**Comment by dlsun** [comment]
I believe this issue is addressed by #79

**Comment by kevindavisross** [comment]
Yes, I'll close this.
